### DISCUSSION TITLE
Figuring out how to connect an Aquacal HP

### DISCUSSION THREAD

**Discussion by jamestalmage** [tagyoureit/nodejs-poolController#852] [RS485]
I've got my RS-485 communicating to the pump, now trying to figure out my Aquacal heater.

I've got the required HP9 motherboard (earlier Aquacals did not emulate Pentair HP), and according to [this thread](https://www.troublefreepool.com/threads/aquacal-heater-chiller-heat-pump-with-intellicenter.266373/post-2330984), it should be replicating an UltraTemp with cooling enabled.

I have (of course) tried reversing the RS-485 polarity to the HeatPump, still no luck. Aquacal's has [a video](https://www.youtube.com/watch?v=HAIU4bCTc0M) documenting the correct installation method, so it seems unlikely I've got it connected to the wrong port / haven't setup the Aquacal Correctly.

I called Aquacal support, and tried talking through it with them. They guided me through factory-default resets, and setting up the RS-485 mode on the unit from scratch. Still no luck.

I have even tried bypassing the MEGA_BAS RS485 and the other connected equipment, and wired directly to a USB adapter (one I know is good, as I used it to troubleshoot).

My question is this...
When I look at MessageManager or even the output of the `od -x` command, I'm seeing NOTHING incoming from the heater (only outbound messages).

What's a good way to confirm the heater's even talking? That I don't have a bum RS-485 chip on the heater MB? 

Does the heater send anything on the RS-485 line without first receiving a message from the controller? (i.e. even if PoolController is speaking the wrong protocol, should I still see SOME communication from the HP?)

Seems like MessageManager doesn't show bits it doesn't understand, am I correct? Is `od -x` the correct tool to look for incoming bits when we're unsure of the protocol?

**Comment by jamestalmage** [comment]
Should mention, I have an oscilloscope, but wouldn't know what I'm looking at for RS-485

**Reply by rock-crusher** [reply]
Just in case you don’t already have this here is the link for the instructions on what you are doing: 
[AquaCal® Connecting Heat Pump to an External Controller](https://www.aquacal.com/wp-content/uploads/manuals/landingpage/externalcontroller/Content/Resources/Topics_Manuals_QS/pdfversions/Connecting%20Heat%20Pump%20to%20an%20External%20Controller.pdf)
Hope this helps.

**Comment by tagyoureit** [comment]
Just to confirm, when you say you see outbound messages, does that mean you have setup the UltraTemp in Nixie and njsPC is trying to communicate to the HP?  Or do you just have other equipment setup and you are seeing generic/other messages sent out?

**Reply by jamestalmage** [reply]
Yes, njsPC is sending messages.... I'm seeing total radio silence from the Aquacal. I set it to "Smart" controller, which it specifically says is designed to work with Jandy and Pentair automation (I referenced a thread above where someone did indeed get it to work with a Pentair panel and show up as an Ultratemp).

I see nothing in MessageManager indicating any receipt of data relating to the heater (not even junk data). The chlorinator and the pump are communicating fine, and I can see messages in both direction.

I see only a few options:

- There is some baud-rate mismatch preventing communication. (I doubt it. The success connecting to Pentair panels suggests it should work with the defaults).
- It's working on some old Pentair format that njsPC doesn't understand. I'd hope that "MessageManager" would just display packets it doesn't understand, but if it just silently discards bits it doesn't understand, that might explain things.
- The RS-485 module on my 3-month old heat pump is bad out-of-the-box and I need to convince the manufacturer of it.

**Reply by tagyoureit** [reply]
Can you attach a [packet capture](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel)?  We can look at the logs and help you see what's going on.

**Reply by jamestalmage** [reply]
Good grief there are a lot of different parts to this app! Had no idea packet capture was a thing (or message monitor until recently). What's the difference in terms of data captured?

I can't do a packet capture easily tonight, maybe tomorrow or this weekend, I disconnected RS-485 earlier today and just went with the Fireman's switch. Would like cooling control though.

I did screenshot MessageMonitor earlier. Not sure if this will help, but it really does seem like there's nothing coming in. This was with pump and chlorinator disconnected, and deleted in njsPC, so I knew any RS-485 activity was heater related.


<img width="732" alt="Screenshot 2023-09-15 at 5 37 26 PM" src="https://github.com/tagyoureit/nodejs-poolController/assets/4082216/be6db0c1-9846-4ef9-8866-26899d04f510">

**Reply by tagyoureit** [reply]
Are these the [instructions](https://www.aquacal.com/wp-content/uploads/manuals/landingpage/air_source_installer_hp9/content/resources/topics_hp7_hp9_hp10/4%20installation%20hp9/hp11%20connecting%20an%20external%20controller%20-%20smart%20bus.htm?Highlight=485) you are following?

**Reply by jamestalmage** [reply]
Yes. That's the one.

Port B, located exactly as the instructions say. I contacted support to make sure I wasn't misreading which port was correct, and re-traced all the steps with them. Including a factory reset.

I've tested continuity on all wires, swapped polarity, etc.

Like I said, I'm now very confident I have not miswired, or incorrectly configured the Heat-pump. And I know the RS-485 works on other devices.

It's gotta be: Incompatible baud rate or communication protocol (despite being advertised as Ultratemp compatible), or something wrong with the Heaters RS-485 chip.

Does MessageManager show anything if it receives garbage data? Or do I need to do a packet capture for that.

**Reply by tagyoureit** [reply]
Yes, it will show invalid packets.  These will also be captured in the logs if you have the right level of output selected.
